enhancements/cluster-logging/cluster-logging-es-pipeline-processing.md

```diff
@@ -0,0 +1,205 @@
+* An OCP user (or admin only?) has enough information and tooling to undertstand and evaluate the impact of creating additional indices. This specifically means new index level metrics will need to be introduced and relevant alerts must be provided as well. Notice that we have recently turned off index level Prometheus metrics (because they were high-cardinality metrics) thus we will need to revisit index level metrics and provide new aggregated non-high cardinality metrics.
+* Any additional non-default indices must be subject to existing retention policy. At the implementation level this means that any new indices have names that comply with data retention implementation OOTB.
+* A multi-tenant data visibility works seamlessly.
+* A roll-over process can manage additional indices seamlessly.
+* An OCP user is able to query data from additional indices.
+* An OCP user is provided enough documentation to be able to understand **when** and **how** to use this feature.
+
+Optional points, not required for inifial implementation:
+
+* An OCP user (or admin only?) can provide custom index mapping for the data. This is specifically needed if the [default Elasticsearch type inference mechanism](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-field-mapping.html) does not lead to acceptable results.
+
+### Non-Goals
+
+* This proposal does not addresses limitations implemented by Elasticsearch to prevent [mapping explosion](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/mapping.html#mapping-limit-settings) (except we need to provide more metrics if needed, see above).
+* This proposal does not enable storing documents with conflicting mappings into common index.
+
+
+## Proposal
+
+When incoming document contains custom data it can happen that it can not be stored into default index because existing index mapping (the index data schema) would conflict with the data in the incoming document.
+
+The main idea in this proposal is to avoid mapping conflicts by pointing incoming document into diffrent index than the default one (how it is decided is discussed later). If that different index hasn't been created yet then it is created on the fly (this is how Elasticsearch works OOTB). In this porposal this index is called an **additional index**.
+
+When an additional index is created then it can optionally use additional [index templates](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/indices-templates.html) to further specify types of individual custom fields. In the begining we assume that [default Elasticsearch type inference mechanism](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-field-mapping.html) does good job in most cases and we do not focus on additional index templates (let's leave this for the later).
```

Comment:
if we aren't going to solve this with the proposal can we add it to the list of `non-goals` ?

Reply:
Yea, it is now in "optional" section but perhaps we should either say this will be addressed later or explicitly state this is non-goal atm. The main issue holding us back with this now is that we need to discuss how to enable this (who should be responsible for what ... etc, and I do not think we have a clear answer).